### PR TITLE
Feature/#172 chtapal search

### LINKED ISSUE #172: Adding chatpal-search package
https://github.com/redlink-gmbh/chatpal-search

using: as search endpiont
http://localhost:8080/rocket/localhost/search-message

### PR DESCRIPTION
**Pull request by ruKurz**
resolves #172 - Adding chatpal-search package

### CODE REVIEW

**Comment by mrsimpson** [comment]
works on my machine including auth now.
Tomorrow, I'll add the defaulting of the Chatpal-settings on the basis of Smarti-settings.

**Comment by mrsimpson** [comment]
...that was easy. Invalidation and copying implemented. 
@ruKurz Please test

File: packages/assistify-ai/client/chatpalAdoption.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+TAPi18n.loadTranslations({
+	en: {
+		CHATPAL_ENTER_SEARCH_STRING: 'Search knowledge base'
```
Comment (mrsimpson) [inline packages/assistify-ai/client/chatpalAdoption.js]: This was a try overwriting the i18-Definition without modifying it. However, this doesn't seem to work. Kept it for reference though

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "packages/chatpal-search"]
+	path = packages/chatpal-search
+	url = https://github.com/assistify/chatpal-search.git
```
Comment (mrsimpson) [inline .gitmodules]: Added the chatpal repository as subpackage. However, I have no clue what this really means and just hope all is good. All changes of the chatpal have been pushed to this repo

File: packages/assistify-ai/config.js [hunk 3]
```diff
@@ -77,3 +77,26 @@ Meteor.startup(() => {
...
 });
+
+RocketChat.settings.get('Assistify_AI_Smarti_Base_URL', (id, smartiValue) => {
+	const domain = RocketChat.models.Settings.findOneNotHiddenById('Assistify_AI_Smarti_Domain');
+	if (domain) {
+		RocketChat.models.Settings.update({_id: 'CHATPAL_BASEURL'}, {
+			$set: {
+				value: `${ smartiValue }rocket/${ domain.value }/search-message`,
+				readonly: true,
+				enableQuery: '{"_id":"Assistify_AI_Smarti_Base_URL","value":""}'
```
Comment (mrsimpson) [inline packages/assistify-ai/config.js]: This disables the input field if the value is effectively copied from the Assistify-setting

**Comment by ThomasRoehl** [comment]
Chatpal-search is not reloading the settings for the Smarti connection.
I had to restart the RC server in order to get it running.

It would be nice to have a button (maybe the little lamp icon, which is already there) to click for triggering the search, or have a live search like the RC channel search.

**Comment by ThomasRoehl** [comment]
When Smarti has completed Conversations from topics/requests which have been deleted from RC, the results are still shown in the search but without correct titel and username and when you click on the "show conversation" button, RC shows an error page.
screen shot 2017-12-15 at 14 53 30

**Comment by mrsimpson** [comment]
There are functional discussions in https://github.com/assistify/Rocket.Chat/issues/172#issuecomment-352383538 which make it impossible at the current state to merge it

**Comment by mrsimpson** [comment]
@rukurz i‘d like to close this one here and open a new one on 0.61

**Comment by mrsimpson** [comment]
replaced by #210 - closing this one

**Pull request closed without merging**